test/datadog/apm_instrumentation_test.go

```diff
@@ -0,0 +1,172 @@
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/namespaces_and_targets.yaml",
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "both libversions and targets",
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/libversions_and_targets.yaml",
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "both enabled and disabled namespaces",
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/enabled_and_disabled_namespaces.yaml",
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "both matchLabels and matchNames for namespace selector",
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/namespace_labels_and_names.yaml",
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "both matchExpressions and matchNames for namespace selector",
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/namespace_exprs_and_names.yaml",
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "extraneous instrumentation key",
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/extra_instrumentation_key.yaml",
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "extraneous target key",
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/extra_target_key.yaml",
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "extraneous pod selector key",
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/extra_podselector_key.yaml",
+				},
+			},
+			isValid: false,
+		},
+		{
+			name: "extraneous namespace selector key",
+			command: common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
+				Values: []string{
+					"../../charts/datadog/values.yaml",
+					"values/instrumentation/extra_namespaceselector_key.yaml",
+				},
+			},
+			isValid: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
These tests could all be parameterized on a valuesFile and the test creates the chart with default values.